extensions/amp-audio/0.1/test/test-amp-audio.js

```diff
@@ -26,172 +27,132 @@ describes.realWin(
   },
   (env) => {
     let win, doc;
-    let ampAudio;
+    let html;
 
     beforeEach(() => {
       win = env.win;
       doc = win.document;
+      html = htmlFor(doc);
+      naturalDimensions_['AMP-AUDIO'] = {width: '300px', height: '30px'};
     });
 
-    function getAmpAudio(attributes, opt_childNodesAttrs) {
-      ampAudio = doc.createElement('amp-audio');
-      for (const key in attributes) {
-        ampAudio.setAttribute(key, attributes[key]);
-      }
-      if (opt_childNodesAttrs) {
-        opt_childNodesAttrs.forEach((childNodeAttrs) => {
-          let child;
-          if (childNodeAttrs.tag === 'text') {
-            child = doc.createTextNode(childNodeAttrs.text);
-          } else {
-            child = doc.createElement(childNodeAttrs.tag);
-            for (const key in childNodeAttrs) {
-              if (key !== 'tag') {
-                child.setAttribute(key, childNodeAttrs[key]);
-              }
-            }
-          }
-          ampAudio.appendChild(child);
-        });
+    async function attachAndRun(element) {
+      expect(element.tagName).to.equal('AMP-AUDIO');
+      doc.body.appendChild(element);
+      await element.buildInternal();
+      try {
+        await element.layoutCallback();
+      } catch (error) {
+        // Ignore failed to load errors since sources are fake.
+        if (error.toString().indexOf('Failed to load') < 0) {
+          throw error;
+        }
```

Comment:
Can the `Failed to load error` be accounted for via `expectAsyncConsoleError` instead? Test files usually recommend this be at the top of the testing block, not sure what happens if it's called from a helper at the top of the testing block.

Reply:
`expectAsyncConsoleError` confirms that the error is triggered exactly N times, which is not constant across tests. I'd rather do it like this instead of having each test specify how many of its sources will fail.